### PR TITLE
feat(init): add support for profile flag

### PR DESCRIPTION
**Pull request by remyleone**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request.
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Release note for [CHANGELOG](https://github.com/scaleway/scaleway-cli/blob/v2/CHANGELOG.md):

```release-note

```

### CODE REVIEW

File: internal/core/client.go [hunk 1]
```diff
@@ -57,7 +57,9 @@ func createClient(buildInfo *BuildInfo, profileName string) (*scw.Client, error)
 	}
 
 	userAgent := "scaleway-cli/"
-	userAgent += buildInfo.Version.String()
+	if buildInfo.Version != nil {
```
Comment (jerome-quere) [inline internal/core/client.go:60]: Should we have a `GetUserAgent` on the `BuildInfo`  ?

File: internal/core/cobra_utils.go [hunk 2]
```diff
@@ -19,9 +19,11 @@ func cobraRun(ctx context.Context, cmd *Command) func(*cobra.Command, []string)
 		meta := extractMeta(ctx)
 		meta.command = cmd
 
-		// If command require a client we make sure a client is present with proper configuration
-		if !cmd.AllowAnonymousClient {
-			err := validateClient(meta.Client)
+		// If command requires authentication, we create a new client and overwrite the existing one
```
Comment (jerome-quere) [inline internal/core/cobra_utils.go:22]:
```suggestion
		// If command requires authentication and the client was not directly provided in the bootstrap config, we create a new client and overwrite the existing one
```

File: internal/core/testdata/test-interrupt-error-empty-success-json.stdout.golden [hunk 3]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (jerome-quere) [inline internal/core/testdata/test-interrupt-error-empty-success-json.stdout.golden:1]: old golden fil is probably still there

File: internal/core/client.go (hunk 1 above)
Comment (remyleone) [reply internal/core/client.go:60]: Sure

Comment (remyleone) [reply internal/core/client.go]: done

File: internal/core/client.go [hunk 4]
```diff
@@ -56,12 +56,10 @@ func createClient(buildInfo *BuildInfo, profileName string) (*scw.Client, error)
 		}
 	}
 
-	userAgent := "scaleway-cli/"
-	userAgent += buildInfo.Version.String()
 	opts := []scw.ClientOption{
 		scw.WithDefaultRegion(scw.RegionFrPar),
 		scw.WithDefaultZone(scw.ZoneFrPar1),
-		scw.WithUserAgent(userAgent),
+		scw.WithUserAgent(buildInfo.GetUserAgent("scaleway-cli")),
```
Comment (jerome-quere) [inline internal/core/client.go:62]: "scaleway-cli" should be handled by the GetUserAgement method

File: internal/core/client.go [hunk 5]
```diff
@@ -73,6 +71,21 @@ func createClient(buildInfo *BuildInfo, profileName string) (*scw.Client, error)
 	return client, nil
 }
 
+func createAnonymousClient(buildInfo *BuildInfo) (*scw.Client, error) {
+	opts := []scw.ClientOption{
+		scw.WithDefaultRegion(scw.RegionFrPar),
+		scw.WithDefaultZone(scw.ZoneFrPar1),
+		scw.WithUserAgent(buildInfo.GetUserAgent("scaleway-cli-anonymous")),
```
Comment (jerome-quere) [inline internal/core/client.go:78]: UserAgent should not be used to store information about the authentication

**Pull request merged**